### PR TITLE
Puppet 4 future proof

### PR DESCRIPTION
**Pull request by acidprime**
Pulling in the Changes needed for the latest version of Puppet Enterprise

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -65,7 +67,7 @@
 
       # Puppet Enterprise 3.8 and ships an embedded r10k so thats all thats supported
       # This conditional should not effect FOSS customers based on the fact 
-      unless ($::is_pe == 'true' or $::is_pe == true) and versioncmp($::pe_version, '3.8.0') >= 0 {
+      unless $is_pe_server and versioncmp($::puppetversion, '3.8.0') >= 0 {
```
Comment (ghoneycutt) [inline manifests/install.pp]: suggest using `if !` as `unless` is confusing

File: manifests/params.pp [hunk 2]
```diff
@@ -57,8 +57,38 @@
     $functions_path     = '/etc/rc.d/init.d/functions'
     $start_pidfile_args = '--pidfile $pidfile'
   }
+  
   if $::is_pe == true or $::is_pe == 'true' {
-    # Puppet Enterprise specific settings
+    # < PE 4
+    $is_pe_server      = true
+  }elsif is_function_available('pe_compiling_server_version') {
```
Comment (ghoneycutt) [inline manifests/params.pp]: why the extra check here? is `is_pe` not to be trusted?

Comment (acidprime) [inline manifests/params.pp]: @ghoneycutt its going away as there is no such thing as PE in Puppet 4 on the agent side as its an all in one installer now

Comment (ghoneycutt) [inline manifests/params.pp]: Interesting.. so there will be no difference between the FOSS and PE packages and no need to tell the difference from that standpoint? If there is no reason for `is_pe` anymore, why do you bother setting it in this code? Apologies if my comments are getting sidelined from the PR and I appreciate your explanations of the pending fact changes.

**Comment by acidprime** [comment]
Closing in liue https://github.com/acidprime/r10k/pull/224

**Pull request closed without merging**